packages/web/src/pages/artist-dashboard-page/store/sagas.ts

```diff
@@ -167,11 +168,11 @@ function* pollForBalance() {
   const pollingFreq = remoteConfigInstance.getRemoteVar(
     IntKeys.DASHBOARD_WALLET_BALANCE_POLLING_FREQ_MS
   )
-  const chan = yield* call(doEvery, pollingFreq || 1000, function* () {
+  const chan = (yield* call(doEvery, pollingFreq || 1000, function* () {
     yield* put(getBalance())
-  })
+  })) as unknown as EventChannel<any>
   yield* take(dashboardActions.reset.type)
-  ;(yield* chan).close()
+  chan.close()
```

Comment:
cc @raymondjacobson

Reply:
switching things to ts is scary! glad it's fixed now though. ty so much

Reply:
I debated whether we should even be doing this in a saga. I think it would make sense to use a hook that mounts/unmounts with the components and polls while the component is active. But I was going for minimal change to fix the issue since it's unrelated to this PR